cmd/controller-manager/app/server.go

```diff
@@ -114,6 +114,37 @@ func NewControllerManagerCommand() *cobra.Command {
 	return cmd
 }
 
+func Run(s *options.KubeSphereControllerManagerOptions, configCh <-chan controllerconfig.Config, ctx context.Context) error {
+	ictx := controllerconfig.Context{}
+	ictx.RenewContext(context.TODO())
+	errCh := make(chan error)
+	defer close(errCh)
+	go func() {
+		if err := run(s, ictx.GetContext()); err != nil {
+			errCh <- err
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			ictx.CancelContext()
+			return nil
+		case cfg := <-configCh:
+			ictx.CancelContext()
+			s.MergeConfig(&cfg)
```

Comment:
Do we need to validate the new config before merging? Or just merge to a fast fail.

Reply:
When misconfigured, I think the app should just crash directly